public/app/plugins/gen.go

```diff
@@ -33,6 +36,116 @@ var skipPlugins = map[string]bool{
 const sep = string(filepath.Separator)
 
 func main() {
+	if len(os.Args) > 1 {
+		log.Fatal(fmt.Errorf("plugin thema code generator does not currently accept any arguments\n, got %q", os.Args))
+	}
+
+	cwd, err := os.Getwd()
+	if err != nil {
+		log.Fatal(fmt.Errorf("could not get working directory: %s", err))
+	}
+	grootp := strings.Split(cwd, sep)
+	groot := filepath.Join(sep, filepath.Join(grootp[:len(grootp)-3]...))
+	lib := cuectx.GrafanaThemaRuntime()
+
+	pluginKindGen := codejen.JennyListWithNamer(func(decl *codegen.PluginTreeDeclForGen) string {
+		return decl.Tree.RootPlugin().Meta().Id
+	})
+
+	outputFile := filepath.Join("pkg", "plugins", "pfs", "corelist", "corelist_load_gen.go")
+	pluginKindGen.Append(codegen.PluginTreeListJenny(outputFile),
+		codejen.AdaptOneToMany(codegen.FlattenJenny(corecodegen.GoTypesJenny("pkg/tsdb", nil)), mapToDecls))
```

Comment:
What's the motivation for the `FlattenJenny`? If it's just flattening individual `File` into a `Files`, `JennyList` already does internally.

Reply:
[You already are](https://github.com/grafana/grafana/pull/58901/files#diff-63ed6c0c1d3b6c1b6fc705eb25d8bbbb1e8834b01ea52939d0410feaf392b50dR54)! It does all the flattening internally.

(I interpret this as a docs deficiency for `JennyList` explaining itself, what it does)